src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -373,6 +373,18 @@ private static BicepCompletionContextKind GetDeclarationTypeFlags(IList<SyntaxBa
                 SyntaxMatcher.IsTailMatch<ModuleDeclarationSyntax, StringSyntax, Token>(matchingNodes, (_, _, token) => token.Type == TokenType.StringComplete) ||
                 SyntaxMatcher.IsTailMatch<ModuleDeclarationSyntax, SkippedTriviaSyntax, Token>(matchingNodes, (module, skipped, _) => module.Path == skipped))
             {
+                if (matchingNodes.Count == 4 &&
+                    matchingNodes[^1] is Token token &&
+                    token.Type == TokenType.StringComplete)
+                {
+                    string text = token.Text;
+
+                    if (text.StartsWith("'br/") || text.StartsWith("'br:"))
+                    {
+                        return BicepCompletionContextKind.OciModuleRegistryReference;
```

Comment:
>OciModuleRegistryReference [](http://example.com/codeflow?start=58&length=26)

Could you move the logic that checks if it's an OCI reference to the completion provider. In general, the completion contexc is meant to figure out what the current cursor position represents in terms of the syntax tree and we let the completion provider use the context to figure out more details and serve the most appropriate completions. 

(This will also avoid us having to add completion context kinds for every possible registry type we have.)

Reply:
Updated here:
https://github.com/Azure/bicep/commit/a699dbfc721de8ace5d95d935c541cfa51959a35